### PR TITLE
The player can send a message with a length of 512 characters

### PR DESCRIPTION
**Pull request by arfush**
## Introduction
PMMP discards messages longer than 255 characters, but the client can send up to 512 characters. I'm sure the limit should be increased to 512.

### Behavioural changes
Message character limit raised to 512

### CODE REVIEW

**Comment by dktapps** [comment]
I'm not really sure what benefit this has. All I can see is that now players can send twice as much spam in the same timeframe.

**Comment by arfush** [comment]
> I'm not really sure what benefit this has. All I can see is that now players can send twice as much spam in the same timeframe.

I think that for this, developers should write their own filters.

Personally, I am concerned that my players cannot send messages longer than 255 characters, and cannot even find out the reasons.

File: src/player/Player.php [hunk 1]
```diff
@@ -1292,7 +1292,7 @@ public function chat(string $message) : bool{
 
 		$message = TextFormat::clean($message, false);
 		foreach(explode("\n", $message) as $messagePart){
-			if(trim($messagePart) !== "" and strlen($messagePart) <= 255 and $this->messageCounter-- > 0){
+			if(trim($messagePart) !== "" and strlen($messagePart) <= 512 and $this->messageCounter-- > 0){
```
Comment (dktapps) [inline src/player/Player.php:1295]: You need to use `mb_strlen()` for correct results. If sending Unicode chars, e.g.:
```
∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏∏
```
This is 512 chars, but 1536 bytes.

Comment (arfush) [reply src/player/Player.php:1295]: I'm sorry for my stupid mistake. Do I have to create a new PR or does GitHub allow me to add a commit to an already existing PR? Sorry, I'm only superficially familiar with this system

Comment (Mohagames205) [reply src/player/Player.php:1295]: You can simply add a commit to your branch on your repository.

Comment (arfush) [reply src/player/Player.php:1295]: OK. Thanks you.
I do it today

**Comment by dktapps** [comment]
Closing as obsolete due to lack of activity (implemented independently by 8943d8a2a7d13c6adcdc11346b461a55e8a88363).

**Pull request closed without merging**